scipy-1.0/minimize.tex

```diff
@@ -0,0 +1,118 @@
+    with the word \textit{Approx.} it means that, while the algorithm does not requires the second-order derivatives from
+    the user, it computes an approximation internally and uses it to accelerate the method convergence.
+    \textit{Local convergence} gives
+    a lower bound to the rate of convergence of the iterations sequence once the iterate is close enough to the
+    objective function solution. The convergence rates mentioned at the table
+    are, from the slowest to the fastest:  linear (L), superlinear (S) and quadratic (Q). The \textit{global convergence}
+    is marked for the algorithms for which there are some proof it always converge to some stationary
+    point (i.e. a point $x^*$ for which $\nabla f(x^*) = 0$).}
+  \label{tab:minimize-unconstrained}
+\end{table}
+
+
+The \texttt{minimize} function provides methods for solving general non-linear optimization
+problems. It unifies several methods under the same general interface in order to facilitate
+interchanging between solvers with different characteristics.  Table~\ref{tab:minimize-unconstrained} summarizes
+all the methods available for solving unconstrained minimization problems of the type:
+\begin{equation}
+  \label{eq:unconstrained-minimization-prob}
+  \min_x f(x)
+\end{equation}
+where $f$ is a multivariate function $f: \mathbb{R}^n \rightarrow \mathbb{R}$ and $x \in \mathbb{R}^n$.
+
+There are a sufficient variety of methods to enable the user to solve problems with the diverse properties.
+Derivative-free methods (e.g. \texttt{Nelder-Mead} e \texttt{Powell}) are appropriate for minimizing noisy objective
+functions. When the derivatives are available, on the other hand, methods with faster convergence can also be used.
+
+There are algorithms implementing two competing strategies: line-search and trust-region. The so-called line-search (LS)
+methods choose a direction and then find the step length that gives the (approximately) optimal point in the given
+direction. Trust-region (TR) methods, on the other hand, fixes the maximum step length for which the local approximation
+is valid and finds approximately the best point inside this \textit{trust-region}.
+
+An example of trust-region method introduced in the latest releases is \texttt{trust-exact}, which a  provides a fast convergence rate by solving 
+ almost exactly the trust-region subproblem. Its downside is it also require the Hessian
+to be stored and factored every iteration, which may renders impossible the solution of
+very large problems.
+
+Some available unconstrained methods are appropriate for large-scale problems
+(i.e. problems with more than 10000 variables).
+\texttt{CG} method does not use second derivatives and is adequate to large-scale problems due to the low storage requirements.
+\texttt{Newton-CG}, \texttt{trust-ncg} and \texttt{trust-krylov} use the second order derivatives in order to get faster
```

Comment:
"use the second order derivatives in order to get faster"->"use second order derivatives to achieve faster"

Reply:
ok